src/applications/app.helpers.js

```diff
@@ -36,15 +36,53 @@ export function isntLoaded(app) {
 
 export function shouldBeActive(app) {
   try {
-    return app.activeWhen(window.location);
+    if (Array.isArray(app.activeWhen))
+      return app.activeWhen.some(conditionForAppActivationMet);
+    return conditionForAppActivationMet(app.activeWhen);
   } catch (err) {
     handleAppError(err, app, SKIP_BECAUSE_BROKEN);
   }
+
+  function conditionForAppActivationMet(pathPrefixOrActivityFn) {
+    const { location } = window;
+    if (typeof pathPrefixOrActivityFn === "string") {
+      const hashRouting = pathPrefixOrActivityFn.startsWith("/#/");
```

Comment:
Totally forgot... Since we are on the subject, I personally dislike having hashrouting in a url other than root, based on how the url appears malformed (for normal internet users), and how it might confuse people that 2 levels of routing is happennig! 
Do you have some thoughts on this? I understand that companies with legacy multpage apps, can use this strategy so they can migrating page by page to a SPA solution, but is this the only use-case or am I missnig something?
Ultimate question, which one is better? hashRouter or (as react-router coins this term) BrowserRouter?

Reply:
> Do you have some thoughts on this? I understand that companies with legacy multpage apps, can use this strategy so they can migrating page by page to a SPA solution, but is this the only use-case or am I missnig something?
Ultimate question, which one is better? hashRouter or (as react-router coins this term) BrowserRouter?

Hash routing is completely valid - single-spa should not take a stance on which browser routing features are good or bad. We should support however people need to do their routing - might be due to migration and might be due to preference.